lib/datadog/appsec/monitor/gateway/watcher.rb

```diff
@@ -47,6 +53,30 @@ def watch_user_id(gateway = Instrumentation.gateway)
                 stack.call(user)
               end
             end
+
+            def watch_user_login(gateway = Instrumentation.gateway)
+              gateway.watch('appsec.events.user_lifecycle', :appsec) do |stack, kind|
+                context = AppSec.active_context
+
+                next stack.call(kind) unless WATCHED_LOGIN_EVENTS.include?(kind)
+
+                persistent_data = { "server.business_logic.#{kind}" => ARBITRARY_VALUE }
+                result = context.run_waf(persistent_data, {}, Datadog.configuration.appsec.waf_timeout)
+
+                if result.match? || !result.derivatives.empty?
```

Comment:
can we use `result.derivatives.any?` here to avoid negation? Or `result.derivatives` might not be an array?